lms/services/application_instance.py

```diff
@@ -35,6 +36,85 @@ def __init__(self, application_instance: ApplicationInstance):
         )
 
 
+@dataclass
+class SearchBuilder:
+    base_model: BASE
+    clause_builders: dict
+    key_map: dict = field(default_factory=lambda: {"id_": "id", "type_": "type"})
+    query_start: Optional[callable] = None
```

Comment:
The defaults are all chosen so you can get a pretty effective search helper with minimal setup:

```python
orgs = SearchBuilder(Organization).get_query(
    db_session, 
    {"id_": 4, "public_id": "us.lms.org.BLAH"}, 
    combine=sa.or_
).all()
```